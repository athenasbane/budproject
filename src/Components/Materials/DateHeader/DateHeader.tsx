import React from 'react';
import { ITransaction } from '../../../types/Interfaces/Api.interface';
import { Container, Title } from '../../Atoms';
import ExpenseItem from '../../Molecules/ExpenseItem/ExpenseItem';

interface IProps {
    expense: ITransaction;
}

const DateHeader = ({ expense }: IProps) => {
    return (
        <Container direction="column">
            <Title textAlign="left" color="white">
                {expense.date.split('-').reverse().join('/')}
            </Title>
            <ExpenseItem {...expense} key={expense.id} />
        </Container>
    );
};

export default DateHeader;
