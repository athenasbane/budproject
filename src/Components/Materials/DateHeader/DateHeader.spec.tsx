import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import DateHeader from './DateHeader';
import { mockExpense } from '../../../__mocks__/expense.mock';

describe('<DateHeader />', () => {
    it('should match snapshot', () => {
        const tree = renderer.create(<DateHeader expense={mockExpense} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
