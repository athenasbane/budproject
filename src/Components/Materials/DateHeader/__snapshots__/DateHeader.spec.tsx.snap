// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateHeader /> should match snapshot 1`] = `
<div
  className=""
  direction="column"
>
  <h1
    className=""
    color="white"
  >
    30/06/2018
  </h1>
  <article
    className=""
  >
    <div
      className=""
      direction="row"
    >
      <h1
        className=""
      >
        T
      </h1>
      <div
        className=""
        direction="column"
      >
        <h1
          className=""
        >
          Tesco
        </h1>
        <h1
          className=""
        >
          Groceries
        </h1>
      </div>
    </div>
    <div
      className=""
    >
      <h1
        className=""
      >
        Â£
        57.21
      </h1>
    </div>
  </article>
</div>
`;
